jobs/build/promote/Jenkinsfile

```diff
@@ -327,25 +333,42 @@ node {
-                                    name: 'IN_FLIGHT_PREV',
-                                ),
-                                string(
-                                    defaultValue: "${suggest_previous}",
-                                    description: (acquire_failure?acquire_failure:"Doozer thinks these are the other releases to include.") + " Edit as necessary (comma delimited).",
-                                    name: 'SUGGESTED',
-                                ),
-                            ]
-                        )
+                    in_flight_prev_required = true
+                    if (params.IN_FLIGHT_PREV) {
+                        in_flight_prev = params.IN_FLIGHT_PREV.trim()
+                        pattern = /4\.$prevMinor\.(\d+)/
+                        match = in_flight_prev =~ pattern
+                        match.find()
+                        if (match.size() == 1) {
+                            print("Proceeding with given in_flight_prev: $in_flight_prev")
+                            in_flight_prev_required = false
+                        } else {
+                            print("Error validating given in_flight_prev: $in_flight_prev . Asking for manual input")
+                        }
+                    } 
 
-                        previousList = commonlib.parseList(resp.SUGGESTED) + commonlib.parseList(resp.IN_FLIGHT_PREV)
+                    if (in_flight_prev_required) {
+                        commonlib.inputRequired(taskThread) {
+                            def resp = input(
+                                message: "${acquire_failure}What PREVIOUS releases should be included in ${release_name} (arch: ${arch})?",
+                                parameters: [
+                                    string(
+                                        defaultValue: "4.${prevMinor}.?",
+                                        description: "This is release ${release_name}. What release is in flight for the previous minor release 4.${prevMinor}?",
+                                        name: 'IN_FLIGHT_PREV',
+                                    ),
+                                    string(
+                                        defaultValue: "${suggest_previous}",
+                                        description: (acquire_failure?acquire_failure:"Doozer thinks these are the other releases to include.") + " Edit as necessary (comma delimited).",
+                                        name: 'SUGGESTED',
+                                    ),
+                                ]
+                            )    
+                        }
+                        in_flight_prev = resp.IN_FLIGHT_PREV
                     }
+                    
+                    previousList = commonlib.parseList(resp.SUGGESTED) + commonlib.parseList(in_flight_prev)
```

Comment:
If `in_flight_prev_required` is false, `resp.SUGGESTED` here will became undefined. I think we still need to promote the artist for the suggested previous unless we have a good way to get rid of that.

Reply:
Thanks for catching that, fixed it